Userland/Utilities/chmod.cpp

```diff
@@ -67,14 +66,15 @@ int main(int argc, char** argv)
     Mask mask;
 
     /* compute a mask */
-    if (argv[1][0] >= '0' && argv[1][0] <= '7') {
-        if (sscanf(argv[1], "%ho", &mask.get_applying_mask()) != 1) {
+
+    if (arguments.strings[1][0] >= '0' && arguments.strings[1][0] <= '7') {
+        if (sscanf(arguments.strings[1].characters_without_null_termination(), "%ho", &mask.get_applying_mask()) != 1) {
             perror("sscanf");
             return 1;
         }
         mask.get_removal_mask() = ~mask.get_applying_mask();
     } else {
-        const char* access_string = argv[1];
+        const char* access_string = arguments.strings[1].characters_without_null_termination();
 
         while (*access_string != '\0') {
```

Comment:
We should loop on the length of the StringView instead of assuming it t will be null terminated (it will be... But it's a bug to assume that with the StringView API)

Reply:
I was trying to avoid changing too much of the implementation logic since the bugs are probably going to hunt me into next year.

Would something like this be acceptable?
```cpp
const char* access_string = arguments.strings[1].to_string().characters();
```

Reply:
Struggled a bit with this. Ended up with the code below since the while loop would not recognize the contents with the one I proposed above.
```cpp
String temp = arguments.strings[1].to_string();
const char* access_string = temp.characters();
```